openassessment/xblock/openassessmentblock.py

```diff
@@ -1106,11 +1106,23 @@ def is_closed(self, step=None, course_staff=None):
             datetime.datetime(2015, 3, 27, 22, 7, 38, 788861)
 
         """
-        submission_range = (self.submission_start, self.submission_due)
-        assessment_ranges = [
-            (asmnt.get('start'), asmnt.get('due'))
-            for asmnt in self.valid_assessments
-        ]
+        if self.due is not None \
+           and self.submission_due is not None \
+           and parse_date_value(self.due, self._) > parse_date_value(self.submission_due, self._):
```

Comment:
This is hard to read. Can we refactor it?
```suggestion
        due_date_greater = parse_date_value(self.due, self._) > parse_date_value(self.submission_due, self._):
        if self.due and self.submission_due and due_date_greater
```